src/main.rs

```diff
@@ -73,52 +77,53 @@ fn three_letter_code(aa: char) -> String {
     ]
     .iter()
     .copied()
-    .collect();
+    .collect()
+});
 
+/// get the three-letter equivalent of an amino acid
+/// aa (char): the amino acid to translate, eg 'A' -> "Ala"
+/// returns: String
+fn three_letter_code(aa: char) -> Result<String, Error> {
     // check input.
-    let re = Regex::new(r"[A-Z*]").unwrap();
-    if !re.is_match(&aa.to_string()) {
-        println!("Invalid amino acid: {}", aa);
-        process::exit(1);
+    match aa {
+        'A'..='Z' => Ok(THREE_LETTER_CODE[&aa].to_string()),
+        _ => Err(Error::InvalidAminoAcid(aa)),
```

Comment:
There is nothing wrong with using the `Regex` crate here, but since `aa` is always a single character and it is checking to see if it is in the `A-Z` range, we can use a `match` with the `'A'..='Z'` range to verify if it is valid.

I also changed the return type of the function to return `Result<String, Error>`. This means that now we can return an error to whoever is calling the function, instead of finishing the process. It is still possible to capture this error and exit with `process::exit(1)`, but this is a decision of the caller of the function (treat the error and exit? recover from it somehow?)